etc/httpd/global.conf

```diff
@@ -936,9 +927,20 @@ RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/features(/?)$ /b/$1thunderbird/f
 RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/features/email_providers\.html$ /$1thunderbird/email-providers/ [L,R=301]
 RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/email-providers(/?)$ /b/$1thunderbird/email-providers$2 [PT]
 
-# bug 1123399
-RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/all\.html?$ /$1thunderbird/all/ [L,R=301]
-RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/all(/?)$ /b/$1thunderbird/all$2 [PT]
+# bug 1123399, 1150649
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/all\.html$ /$1thunderbird/all/ [L,R=301]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/all-beta\.html$ /$1thunderbird/beta/all/ [L,R=301]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/((?:beta|earlybird)/)?(all|notes|system-requirements)(/?)$ /b/$1thunderbird/$2$3$4 [L,PT]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/channel(/?)$ /b/$1thunderbird/channel$2 [L,PT]
+
+# bug 1081917, 1029829, 1029838
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/([3-9]\d\.\d(?:a2|beta|\.\d)?)/releasenotes(/)?$ /b/$1thunderbird/$2/releasenotes$3 [L,PT]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/([3-9]\d\.\d(?:a2|beta|\.\d)?)/system-requirements(/)?$ /b/$1thunderbird/$2/system-requirements$3 [L,PT]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/releases(/)?$ /b/$1thunderbird/releases$2 [L,PT]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/releases/(.*)$ http://website-archive.mozilla.org/www.mozilla.org/thunderbird_releasenotes/en-US/thunderbird/releases/$2 [L,R=301]
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/latest/(.*)/?$ /b/$1thunderbird/latest/$2/ [L,PT]
```

Comment:
I think you can leave off the `/?` at the end of the match and the `/` at the end of the destination. There could be some mismatch there that may make for some strange URLs.